torch/_inductor/comm_analysis.py

```diff
@@ -0,0 +1,232 @@
+import math
+from enum import Enum
+
+import torch
+from . import ir
+
+from .utils import get_dtype_size, sympy_product
+from .virtualized import V
+
+
+class NCCL_COLL(Enum):
+    ALL_REDUCE = 0
+    ALL_GATHER = 1
+    REDUCE_SCATTER = 2
+
+
+class NCCL_HW(Enum):
+    NVLINK = 0
+    PCI = 1
+    NET = 2
+
+
+class NCCL_ALGO(Enum):
+    TREE = 0
+    RING = 1
+
+
+class NCCL_PROTO(Enum):
+    SIMPLE = 0
+    LL = 1
+    LL128 = 2
+
+
+class NVIDIA_GPU_TYPE(Enum):
+    VOLTA = 0
+    AMPERE = 1
+    HOPPER = 2
+
+
+# Latencies in us
+# len(NCCL_ALGO) x len(NCCL_PROTO)
+baseLat = torch.tensor(
+    [
+        # Tree
+        [
+            6.8,  # LL
+        ],
+        # Ring
+        [
+            6.6,  # LL
+        ],
+    ]
+)
+
+# Latencies in us
+# len(NCCL_HW) x len(NCCL_ALGO) x len(NCCL_PROTO)
+hwLat = torch.tensor(
+    [
+        # NVLINK
+        [
+            [0.6],  # Tree (LL)
+            [0.6],  # Ring (LL)
+        ],
+        # PCI
+        [
+            [1.0],  # Tree (LL)
+            [1.0],  # Ring (LL)
+        ],
+        # NET
+        [
+            [5.0],  # Tree (LL)
+            [2.7],  # Ring (LL)
+        ],
+    ]
+)
+
+
+# LL128 max BW per channel
+llMaxBws = torch.tensor(
+    [
+        # Volta-N1/Intel-N2/Intel-N4
+        [
+            39.0,
+            39.0,
+            20.4,
+        ],
+        # Ampere-N1/AMD-N2/AMD-N4
+        [
+            87.7,
+            22.5,  # avg of ring & tree
+            19.0,
+        ],
+        # Hopper-N1/AMD-N2/AMD-N4
+        [
+            87.7,
+            22.5,  # avg of ring & tree
+            19.0,
+        ],
+    ]
+)
+
+
+def get_gpu_type() -> NVIDIA_GPU_TYPE:
+    gpu_info = torch.utils.collect_env.get_gpu_info(torch.utils.collect_env.run)
```

Comment:
I don't think `torch.utils.collect_env` works -- it doesn't typecheck and the module lookup doesn't work at runtime either. I guess we probably need to add an `import collect_env` in `__init__.py`

Reply:
thanks for the catch and you are right. Fix coming..